app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php

```diff
@@ -396,9 +396,28 @@ protected function _getAllProductImages($product, $storeId)
      * Get media config
      *
      * @return \Magento\Catalog\Model\Product\Media\Config
+     * @deprecated No longer used, as we're getting full image URL from getProductImageUrl method
      */
     protected function _getMediaConfig()
     {
         return $this->_mediaConfig;
```

Comment:
Looks like you have eliminated all of the usages for this dependency.
Now you could also deprecate the property and remove all type hints to it.